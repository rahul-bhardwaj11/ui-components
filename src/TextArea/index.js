import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AntInput from 'antd/lib/input';
import 'antd/lib/input/style/index.css';
import theme from '../styles/theme';
import styled from 'styled-components';

const MtTextArea = styled.div`
  .ant-input {
    border: 1px solid ${theme.colors.ALTO};
    &:hover,
    &:active,
    &:focus {
      border: 1px solid ${theme.colors.SILVER};
    }

    &:active,
    &:focus {
      box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.08);
    }
  }
`;

const INPUT_TYPES = {
  ALPHA: 'alpha',
  ALPHA_NUMERIC: 'alphaNumeric'
};
class TextArea extends Component {
  static propTypes = {
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    placeholder: PropTypes.string,
    onChange: PropTypes.func,
    type: PropTypes.oneOf([INPUT_TYPES.ALPHA, INPUT_TYPES.ALPHA_NUMERIC])
  };

  static defaultProps = {
    onChange: () => {},
    type: INPUT_TYPES.ALPHA_NUMERIC
  };

  constructor(props) {
    super(props);
    this.state = {
      value: this.props.value || this.props.defaultValue
    };
  }

  validator = value => {
    const { type } = this.props;
    if (type == INPUT_TYPES.ALPHA) {
      const match = value.match(/[^a-zA-Z_ ]/g);
      return match == null ? true : false;
    }
    return true;
  };

  onChange = event => {
    const { onChange } = this.props;
    const value = event.target.value;
    if (this.validator(value)) {
      this.setState({ value });
      onChange(event, value);
    }
  };

  componentWillReceiveProps(newProps) {
    const { value } = this.props;
    let { value: newValue } = newProps;
    if (newValue !== value) {
      this.setState({ value: newValue });
    }
  }

  render() {
    const { value } = this.state;
    return (
      <MtTextArea>
        <AntInput.TextArea
          {...this.props}
          value={value}
          onChange={this.onChange}
        />
      </MtTextArea>
    );
  }
}
export default TextArea;
