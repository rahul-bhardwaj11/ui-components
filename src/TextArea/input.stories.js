import React from 'react';

import { storiesOf } from '@storybook/react';
import TextArea from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, text, number } from '@storybook/addon-knobs';

const stories = storiesOf('TextArea', module);
stories.addDecorator(withKnobs);

stories
  .add(
    'Default TextArea',
    withInfo('Basic usage height based on content lines')(() => (
      <TextArea placeholder={text('Placeholder')} autosize />
    ))
  )
  .add(
    'TextArea with min and max rows',
    withInfo(
      'Basic usage Autosize height with minimum and maximum number of lines'
    )(() => (
      <TextArea
        placeholder={text('Placeholder')}
        autosize={{
          minRows: number('minRows', 2),
          maxRows: number('maxRows', 4)
        }}
      />
    ))
  );
