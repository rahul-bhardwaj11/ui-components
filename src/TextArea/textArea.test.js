import React from 'react';
import renderer from 'react-test-renderer';
import TextArea from './index';

describe('TextArea', () => {
  it('matches default snapshot', () => {
    const tree = renderer
      .create(
        <TextArea placeholder="Enter the scenario description. Example: Record a video directly on MindTickle or upload a pre-recorded video." />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
