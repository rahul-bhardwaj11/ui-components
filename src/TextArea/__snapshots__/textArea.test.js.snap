// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea matches default snapshot 1`] = `
<textarea
  className="ant-input"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Enter the scenario description. Example: Record a video directly on MindTickle or upload a pre-recorded video."
  style={Object {}}
/>
`;
